### PR TITLE
APEXMALHAR-2174-S3-ReaderIssue Fixed the S3 reader issue

### PR DESCRIPTION
**Pull request by chaithu14**

### CODE REVIEW

**Comment by chaithu14** [comment]
@yogidevendra Please review and merge.

File: library/src/main/java/com/datatorrent/lib/io/fs/S3BlockReader.java [hunk 1]
```diff
@@ -47,63 +56,144 @@ public S3BlockReader()
...
    * @throws IOException
    */
   @Override
   protected FSDataInputStream setupStream(BlockMetadata.FileBlockMetadata block) throws IOException
   {
-    FSDataInputStream ins = fs.open(new Path(s3bucketUri + block.getFilePath()));
-    ins.seek(block.getOffset());
-    return ins;
+    String filePath = block.getFilePath();
+    if (filePath.startsWith("/")) {
```
Comment (yogidevendra) [inline library/src/main/java/com/datatorrent/lib/io/fs/S3BlockReader.java:107]: Please add comments mentioning why this is needed.

**Comment by yogidevendra** [comment]
Rest looks OK. I will wait for feedback from other community members. We can merge it after 2 days if there is no other feedback.

**Comment by yogidevendra** [comment]
@chaithu14 travis build failed for some reason. Could you please make sure to make this green?

**Pull request merged**